src/sweeps/params.py

```diff
@@ -366,41 +366,25 @@ def to_config(self) -> Dict:
         """Convert a HyperParameterSet to a SweepRun config."""
         return dict([param._to_config() for param in self])
 
-    def convert_runs_to_normalized_vector(self, runs: List[SweepRun]) -> ArrayLike:
-        """Converts a list of SweepRuns to an array of normalized parameter vectors.
-
-        Args:
-            runs: List of runs to convert.
-
-        Returns:
-            A 2d array of normalized parameter vectors.
-        """
-
-        runs_params = [run.config for run in runs]
-        X = np.zeros([len(self.searchable_params), len(runs)])
-
-        for key, bayes_opt_index in self.param_names_to_index.items():
-            param = self.param_names_to_param[key]
-            row = np.array(
-                [
-                    (
-                        param.value_to_int(config[key]["value"])
-                        if param.type == HyperParameter.CATEGORICAL
-                        else config[key]["value"]
-                    )
-                    if key in config
-                    # filter out incorrectly specified runs
-                    else np.nan
-                    for config in runs_params
-                ]
-            )
-
-            X_row = param.cdf(row)
```

Comment:
i think the main thing we need to do to fix this bug is ensure we are not passing any `nan`s or `None`s to `param.cdf` here. we should do some kind of filtering on `row` before it goes to `param.cdf` (I think `np.isfinite` would be good) and then let update the values of `X` accordingly. here we initialize them to zero so we should keep that for backwards compatibility in the case when we dont have a good value for the param. I think this is done in your refactor below as well